### PR TITLE
Bridge UI demo

### PR DESCRIPTION
**Pull request by denalimarsh**
A Bridge UI that enables the transfer of ETH and ERC20 tokens from Ethereum to Substrate via Metamask. The readme has steps for deployment and testing.

After entering a valid ERC20 token's contract address the user is able to approve, then send tokens to the ERC20App smart contract. Allowances/balances are fetched on the first load, then every 10 seconds.

## Initial load
Screen Shot 2020-09-16 at 1 47 02 PM

## ETHApp
Screen Shot 2020-09-16 at 1 46 29 PM

## ERC20App
Screen Shot 2020-09-16 at 1 43 34 PM

## Sidebar
Screen Shot 2020-09-16 at 2 37 34 PM

### CODE REVIEW

File: packages/app-polkadot-ethereum-bridge/src/config.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Bridge applications
+export const APP_ETH_CONTRACT_ADDRESS : string = "0x351eC8441FEb8347bB3B53A4b59ECb7650B0302b";
+export const APP_ERC20_CONTRACT_ADDRESS : string = "0xab6C826447Ec52bD7Fd1aaa19f420eDb7A302c8F";
```
Comment (philipstanislaus) [inline packages/app-polkadot-ethereum-bridge/src/config.tsx:3]: Seems we have no linter in this repository – could you add that?

File: packages/app-polkadot-ethereum-bridge/README.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ nvm install 10.18
 nvm use 10.18
 
 # Remove yarn lock before install deps
-rm ./yarn.lock 
+rm ./yarn.lock
```
Comment (philipstanislaus) [inline packages/app-polkadot-ethereum-bridge/README.md:41]: Why is that needed?

**Review by philipstanislaus — APPROVED** [review]
LGTM!

**Review by vgeddes — CHANGES_REQUESTED** [review]
Looks good! One problem though is that it doesn't build: https://github.com/Snowfork/substrate-ui/pull/12/checks?check_run_id=1123164163

You can fix the Json import errors by following this guide: https://mariusschulz.com/articles/importing-json-modules-in-typescript

I wouldn't know how to fix the other build errors. As an experiment, maybe try basing your work off `master` instead of stable-base?

**Comment by vgeddes** [comment]
Does this PR obsolete #8? I see it is still unmerged.

File: packages/app-polkadot-ethereum-bridge/src/config.tsx (hunk 1 above)
Comment (vgeddes) [reply packages/app-polkadot-ethereum-bridge/src/config.tsx:3]: Turns out there is a linter, but the job was aborted for some reason: https://github.com/Snowfork/substrate-ui/pull/12/checks?check_run_id=1123164092

But yeah, I'm seeing lots of lint errors while I've got the code checked out locally.

File: packages/app-polkadot-ethereum-bridge/README.md (hunk 2 above)
Comment (denalimarsh) [reply packages/app-polkadot-ethereum-bridge/README.md:41]: I repeatedly hit an error while installing deps and found this to be a workable solution. Didn't dig into it too far, so not sure if it's due to legacy issues with the codebase or my configuration. @vgeddes maybe you've already addressed this locally while working with the build process?

**Comment by denalimarsh** [comment]
> Does this PR obsolete #8? I see it is still unmerged.

Yeah, it does. Those changes are included in this PR. Closed it 👍

File: packages/app-polkadot-ethereum-bridge/README.md (hunk 2 above)
Comment (vgeddes) [reply packages/app-polkadot-ethereum-bridge/README.md:41]: I've made the app compatible with the `master` branch in https://github.com/Snowfork/substrate-ui/pull/14. Didn't run into any issues there.

**Comment by vgeddes** [comment]
@denalimarsh lets consider closing this PR in favour of https://github.com/Snowfork/substrate-ui/pull/14. Didn't run into any issues there. I've made it compatible with the `master` branch and it builds cleanly.

**Comment by denalimarsh** [comment]
Closing in favor of https://github.com/Snowfork/substrate-ui/pull/14.

**Pull request closed without merging**